### PR TITLE
[caclmgrd][DualToR] Fix a case where vlan address is not network address for DualToR Active-active configuration

### PR DESCRIPTION
**Pull request by vdahiya12**
This PR fixes #82 where a corner case is not covered, if vlan address of DUT is not network address. 
Basically a src IP is added to the SNAT rule so that only packets originating from ToR with src IP as vlan IP get natted by the rule and change the src IP to LoopBack IP

For multiple vlan IP's only the vlan address coming from configuration is picked as the vlan address and not the network address. 

We consider a case where there are multiple vlan IP's are example in such format :

```
Vlan705                 10.131.113.17/28     up/up         N/A                   N/A
Vlan805                 10.131.111.129/26    up/up         N/A                   N/A

  "VLAN_INTERFACE": {
        "Vlan705|10.131.113.17/28": {},
        "Vlan805|10.131.111.129/26": {},
```
In such a case we expect iptables rules should be :

```
target     prot opt source               destination         
SNAT       all  --  10.131.113.17        10.131.113.19        to:10.212.64.1
SNAT       all  --  10.131.111.129       10.131.111.131       to:10.212.64.1
```

where soc_ip's are such :
```
10.131.111.131, 10.131.113.19
```

Before this change the souce IP of the vlan is not correctly picked in the iptables rule

### CODE REVIEW

File: scripts/caclmgrd [hunk 1]
```diff
@@ -54,8 +54,10 @@ def get_ipv4_networks_from_interface_table(table, intf_name):
             iface_name, iface_cidr = key
             if iface_name.startswith(intf_name):
                 ip_ntwrk = ipaddress.ip_network(iface_cidr, strict=False)
-                if isinstance(ip_ntwrk, ipaddress.IPv4Network):
-                    addresses.append(ip_ntwrk)
+                ip_str = iface_cidr.split("/")[0]
```
Comment (prsunny) [inline scripts/caclmgrd:57]: Could you check if this works if there are secondary IPs in Vlan? Like two or three different IPs in the same Vlan?

**Review by prsunny — COMMENTED** [review]
Please add an example of the issue and config in the description.

**Comment by zjswhhh** [comment]
> Please add an example of the issue and config in the description.

+1

File: scripts/caclmgrd (hunk 1 above)
Comment (vdahiya12) [reply scripts/caclmgrd:57]: Works as intended, added in description

**Comment by vdahiya12** [comment]
> Please add an example of the issue and config in the description.

added

**Review by zjswhhh — APPROVED** [review]
lgtm

**Comment by StormLiangMS** [comment]
@vdahiya12 could you update ADO in the description?

**Comment by vdahiya12** [comment]
> @vdahiya12 could you update ADO in the description?

@StormLiangMS added

**Pull request merged**